src/Test/AbstractIntegrationTestCase.php

```diff
@@ -269,14 +262,27 @@ function (FixerInterface $fixer) {
                 new NullCacheManager()
             );
 
+            Tokens::clearCache();
             $runner->fix();
             $fixedInputCodeWithReversedFixers = file_get_contents($tmpFile);
 
-            $this->assertNotSame(
-                $fixedInputCode,
-                $fixedInputCodeWithReversedFixers,
-                sprintf('Set priorities must be significant. If fixers used in reverse order return same output then the integration test is not sufficient or the priority relation between used fixers should not be set. In "%s".', $case->getFileName())
-            );
+            // If output is different depends on rules order - we need to verify that the rules are ordered by priority.
+            // If not, any order is valid.
+            if ($fixedInputCode !== $fixedInputCodeWithReversedFixers) {
+                $this->assertGreaterThan(
```

Comment:
I might be wrong again but IMHO, if the output is different in reversed order, we always have an issue that should be fixed using priorities.

Since no matter their orders are, fixers are first sorted by priority, the only reason why output is different in reversed order is because 2 fixers shares the same priority.

Shouldn't we just test that no matter the order of fixers, we always have the same output?
I don't want to bother you with me not understanding the code in details but since you asked me to have a look at this PR, I want it to be clear in my mind :)